changefeedccl: fix TestNoBackfillAfterNonTargetColumnDrop

andyyang890 opened the pull request:
This patch fixes `TestNoBackfillAfterNonTargetColumnDrop` so that it
actually checks that no backfills occur after a non-target column drop.

Fixes #150214

Release note: None

andyyang890 committed:
changefeedccl: fix TestNoBackfillAfterNonTargetColumnDrop

This patch fixes `TestNoBackfillAfterNonTargetColumnDrop` so that it
actually checks that no backfills occur after a non-target column drop.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/150215)
<!-- Reviewable:end -->

andyyang890 committed:
changefeedccl: fix test assertion in TestNoStopAfterNonTargetColumnDrop

This patch replaces an incorrect usage of `require.Regexp` in
`TestNoStopAfterNonTargetColumnDrop`.

Release note: None

aerfrei commented on pkg/ccl/changefeedccl/changefeed_test.go:
```diff
@@ -1931,27 +1931,31 @@ func TestNoBackfillAfterNonTargetColumnDrop(t *testing.T) {
 
 		// Open up the changefeed.
-		cf := feed(t, f, `CREATE CHANGEFEED FOR TABLE hasfams FAMILY b_and_c`, args...)
+		// NB: `updated` is needed to make sure we aren't getting an extra backfill
```
Would it be possible to make this check without having specify updated and then strip the timestamps? My thought is that we could specify initial_scan='no' and just remove that first assertion that we get the `hasfams.b_and_c: [0]->{"after": {"b": "b", "c": "c"}}` payload.

aerfrei replied:
I found it a little confusing to specify updated when we won't actually assert anything about it, though the comment will hopefully make this clear. My sense was that this test was specifically about backfills and how they interact with dropped columns so the initial scan wasn't as relevant. Either way works for me.

andyyang890 replied:
Thanks for the feedback. I updated the comment to hopefully be more clear now.

andyyang890 commented:
TFTRs!

bors r=aerfrei,asg0451
